packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -52,119 +62,251 @@ class _DeepLinkPageState extends State<DeepLinkPage>
+              const SizedBox(width: denseSpacing),
+              SearchBar(
+                leading: const Icon(Icons.search),
+                hintText: 'Search a URL, domain or path',
+                onChanged: (value) {
+                  controller.searchContent = value;
+                },
+                constraints: BoxConstraints.tight(
+                  Size(wideSearchFieldWidth, defaultTextFieldHeight),
+                ),
+              ),
+            ],
+          ),
+          const SizedBox(height: denseSpacing),
+          const TabBar(
+            tabs: [
+              Text('Domain view'),
+              Text('Path view'),
+              Text('Single URL view'),
+            ],
+            tabAlignment: TabAlignment.start,
+            isScrollable: true,
+          ),
+          Expanded(
+            child: ValueListenableBuilder<List<LinkData>>(
+              valueListenable: controller.linkDatasNotifier,
+              builder: (context, linkDatas, _) => ValueListenableBuilder<bool>(
+                valueListenable: controller.showSpitScreenNotifier,
+                builder: (context, showSpitScreen, _) => TabBarView(
+                  children: [
+                    _DataTableWithValidationDetails(
+                      tableView: TableView.domainView,
+                      linkDatas: controller.getLinkDatasByDomain,
+                      controller: controller,
+                      showSpitScreen: showSpitScreen,
+                    ),
+                    _DataTableWithValidationDetails(
+                      tableView: TableView.pathView,
+                      linkDatas: controller.getLinkDatasByPath,
+                      controller: controller,
+                      showSpitScreen: showSpitScreen,
+                    ),
+                    _DataTableWithValidationDetails(
+                      tableView: TableView.singleUrlView,
+                      linkDatas: linkDatas,
+                      controller: controller,
+                      showSpitScreen: showSpitScreen,
+                    ),
+                  ],
+                ),
               ),
             ),
-          ],
-        ),
-        const SizedBox(height: denseSpacing),
-        ValueListenableBuilder<List<LinkData>>(
-          valueListenable: controller.linkDatasNotifier,
-          builder: (context, linkDatas, _) => _DataTable(
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _DataTableWithValidationDetails extends StatelessWidget {
+  const _DataTableWithValidationDetails({
+    required this.linkDatas,
+    required this.tableView,
+    required this.controller,
+    required this.showSpitScreen,
+  });
+  final List<LinkData> linkDatas;
+  final TableView tableView;
+  final DeepLinksController controller;
+  final bool showSpitScreen;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: _DataTable(
```

Comment:
wrap with `RoundedOutlinedBorder`. To make this look right with the header, you may need to play with the `show*` parameters.

Reply:
Will update border later according to figma